smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftWriter.kt

```diff
@@ -97,7 +98,15 @@ class SwiftWriter(private val fullPackageName: String, swiftImportContainer: Swi
 
     fun addImport(symbol: Symbol) {
         if (symbol.isBuiltIn || symbol.isServiceNestedNamespace || symbol.namespace.isEmpty()) return
-        addImport(symbol.namespace)
+        if (symbol.isInternalSPI()) {
+            val kind = symbol.getProperty("kind").orElseThrow()
+            addImport(
+                "$kind ${symbol.namespace}.${symbol.name}",
+                internalSPIName = "Internal"
+            )
+        } else {
+            addImport(symbol.namespace)
+        }
     }
```

Comment:
This change allows adding individual type import by symbol & properties saved in that symbol.